File: ProjectEuler/Problem_4/4-Largest_palindrome_product.py
'''Find the largest palindrome made from the product of two 3-digit numbers.'''

largest = 0

def palindrome(text):
    return text == text[::-1]

for i in range(999,900,-1):
    for j in range(999,900,-1):
        if (i*j)%11==0 and palindrome(str(i * j)) and (i*j) > largest:
            largest = i*j

print(largest) # 906609

File: ProjectEuler/Problem_4/4-Largest_palindrome_product.js
// Find the largest palindrome made from the product of two 3-digit numbers.

const palindrome = text => text == [...text].reverse().join('')

let largest = 0

for (let i = 999; i > 900; i--) {
    for (let j = 999; j > 900; j--) {
        if (palindrome(''+(i * j)) && (i * j) > largest)
            largest = i*j    
    }
}

console.log(largest)